astropy/coordinates/baseframe.py

```diff
@@ -1791,27 +1817,34 @@ def position_angle(self, other: BaseCoordinateFrame | SkyCoord) -> Angle:
         >>> c1.position_angle(c3).degree  # doctest: +FLOAT_CMP
         44.995636455344844
         """
-        return position_angle(*self._prepare_unit_sphere_coords(other))
+        return position_angle(*self._prepare_unit_sphere_coords(other, "ignore"))
 
-    def separation(self, other):
+    def separation(
+        self,
+        other: BaseCoordinateFrame | SkyCoord,
+        frame_mismatch: StrictnessControl = "warn",
+    ) -> Angle:
         """
         Computes on-sky separation between this coordinate and another.
 
-        .. note::
-
-            If the ``other`` coordinate object is in a different frame, it is
-            first transformed to the frame of this object. This can lead to
-            unintuitive behavior if not accounted for. Particularly of note is
-            that ``self.separation(other)`` and ``other.separation(self)`` may
-            not give the same answer in this case.
-
         For more on how to use this (and related) functionality, see the
         examples in :doc:`astropy:/coordinates/matchsep`.
 
         Parameters
         ----------
         other : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
             The coordinate to get the separation to.
+        frame_mismatch : {"warn", "ignore", "error"}
+            If the ``other`` coordinates are in a different frame then they
+            will have to be transformed and depending on the details
```

Comment:
Could you clarify what details ? ("details of the transformation" would be enough, provided that's what you mean here)